examples/CustomCollectionView/Sample/MosaicCollectionViewLayout.m

```diff
@@ -0,0 +1,230 @@
+//
+//  MosaicCollectionViewLayout.m
+//  Sample
+//
+//  Created by McCallum, Levi on 11/22/15.
+//  Copyright (c) 2015 Facebook. All rights reserved.
+//
+
+#import "MosaicCollectionViewLayout.h"
+
+@implementation MosaicCollectionViewLayout {
+  NSMutableArray *_columnHeights;
+  NSMutableArray *_itemAttributes;
+  NSMutableDictionary *_headerAttributes;
+  NSMutableArray *_allAttributes;
+}
+
+- (instancetype)init
+{
+  self = [super init];
+  if (self != nil) {
+    self.numberOfColumns = 3;
```

Comment:
Later, we might use UIDevice or UIScreen to adjust the number of columns depending on device and/or orientation.